### ISSUE TITLE
Generate tagged command throws exception

### ISSUE THREAD

**Issue by simpleim** [putyourlightson/craft-blitz#728]
### Bug Report

Getting an exception when trying to generate caches for a specific tag defined in a template file.
Error is thrown when there's a match between the tag being warmed, and a tag in a template.

```
{% do craft.blitz.options({
    cachingEnabled: true,
    trackElements: false,
    trackElementQueries: false,
    cacheDuration: 'PT3H',
    tags: ['calendar', 'warm5']
}) %}
```

Command being run:
`ddev craft  blitz/cache/generate-tagged warm5`
Exception thrown:
`Exception 'yii\base\UnknownPropertyException' with message 'Setting unknown property: putyourlightson\blitz\models\SiteUriModel::cache'`

Stack-trace:
```
Exception 'yii\base\UnknownPropertyException' with message 'Setting unknown property: putyourlightson\blitz\models\SiteUriModel::cache'

in /var/www/html/vendor/yiisoft/yii2/base/Component.php:217

Stack trace:
#0 /var/www/html/vendor/craftcms/cms/src/helpers/App.php(1441): yii\base\Component->__set('cache', NULL)
#1 /var/www/html/vendor/craftcms/cms/src/base/Model.php(84): craft\helpers\App::configure(Object(putyourlightson\blitz\models\SiteUriModel), Array)
#2 /var/www/html/vendor/putyourlightson/craft-blitz/src/helpers/SiteUriHelper.php(469): craft\base\Model->__construct(Array)
#3 /var/www/html/vendor/putyourlightson/craft-blitz/src/console/controllers/CacheController.php(197): putyourlightson\blitz\helpers\SiteUriHelper::getSiteUrisFromTags(Array)
#4 [internal function]: putyourlightson\blitz\console\controllers\CacheController->actionGenerateTagged(Array)
#5 /var/www/html/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#6 /var/www/html/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams(Array)
#7 /var/www/html/vendor/yiisoft/yii2/console/Controller.php(180): yii\base\Controller->runAction('generate-tagged', Array)
#8 /var/www/html/vendor/yiisoft/yii2/base/Module.php(552): yii\console\Controller->runAction('generate-tagged', Array)
#9 /var/www/html/vendor/yiisoft/yii2/console/Application.php(180): yii\base\Module->runAction('blitz/cache/gen...', Array)
#10 /var/www/html/vendor/craftcms/cms/src/console/Application.php(91): yii\console\Application->runAction('blitz/cache/gen...', Array)
#11 /var/www/html/vendor/yiisoft/yii2/console/Application.php(147): craft\console\Application->runAction('blitz/cache/gen...', Array)
#12 /var/www/html/vendor/craftcms/cms/src/console/Application.php(122): yii\console\Application->handleRequest(Object(craft\console\Request))
#13 /var/www/html/vendor/yiisoft/yii2/base/Application.php(384): craft\console\Application->handleRequest(Object(craft\console\Request))
#14 /var/www/html/craft(13): yii\base\Application->run()
#15 {main}
```

We are also seeing the same error on a separate app instance running the exact same packages, but different DB/PHP versions:
```
- PHP version: 8.2.25
- Craft edition & version: Pro 4.12.8
- Database driver & version: MySQL 8.0.32
```

#### Application Info

- PHP version: 8.1.21
- Craft edition & version: Pro 4.12.8
- Database driver & version: MariaDB 10.4.30

#### Installed Plugins

- Amazon S3: 2.2.1
- Batch Actions: 1.3.1
- Blitz: 4.23.3
- Blitz Recommendations: 2.2.1
- Button Box: 4.2.4
- Control Panel CSS: 2.6.0
- Control Panel JS: 2.6.0
- CP Field Inspect: 1.4.4
- Elements Panel: 2.0.0
- Embedded Assets: 4.1.2
- Feed Me: 5.8.2
- Field Manager: 3.0.9
- Formie: 2.1.32
- Hyper: 1.3.3
- Imager X: 4.4.1
- Imager X AWS Serverless Transformer: 2.1.1
- Maps: 4.0.8
- MatrixMate: 2.2.2
- Minify: 4.0.1
- MJML: 2.0.2
- Optimise: 4.1.5
- Palette: 4.1.2
- Quick Field: 2.0.8
- Redactor: 3.1.0
- Retour: 4.1.19
- Simple: dev-develop
- Smith: 2.0.1
- Super Table: 3.0.15
- Table Maker: 4.0.14
- Twigpack: 4.0.0
- Typogrify: 4.0.4
- UI Element fields: 2.0.3
- Vite: 4.0.10
- Wordsmith: 4.5.0

#### Loaded Modules

- closure: nystudio107\closure\Closure
- codeeditor: nystudio107\codeeditor\CodeEditor
- sprig-core: putyourlightson\sprig\Sprig
- blitz-hints: putyourlightson\blitzhints\BlitzHints
- verbb-base: verbb\base\Base
- debug: craft\debug\Module

#### Blitz Plugin Settings

```json
{
    "debug": false,
    "hintsEnabled": true,
    "cachingEnabled": true,
    "refreshCacheEnabled": true,
    "refreshMode": 1,
    "includedUriPatterns": [
        {
            "siteId": "",
            "uriPattern": ".*"
        },
        {
            "siteId": "",
            "uriPattern": ""
        }
    ],
    "excludedUriPatterns": [],
    "cacheStorageType": "putyourlightson\\blitz\\drivers\\storage\\FileStorage",
    "cacheStorageSettings": {
        "folderPath": "@webroot\/cache\/blitz",
        "compressCachedValues": ""
    },
    "cacheStorageTypes": [],
    "cacheGeneratorType": "putyourlightson\\blitz\\drivers\\generators\\LocalGenerator",
    "cacheGeneratorSettings": {
        "concurrency": 3
    },
    "cacheGeneratorTypes": [],
    "customSiteUris": [],
    "cachePurgerType": "putyourlightson\\blitz\\drivers\\purgers\\DummyPurger",
    "cachePurgerSettings": [],
    "cachePurgerTypes": [],
    "deployerType": "putyourlightson\\blitz\\drivers\\deployers\\DummyDeployer",
    "deployerSettings": [],
    "deployerTypes": [],
    "ssiEnabled": false,
    "ssiTagFormat": "<!--#include virtual=\"{uri}\" -->",
    "detectSsiEnabled": true,
    "esiEnabled": false,
    "queryStringCaching": 0,
    "includedQueryStringParams": [
        {
            "siteId": "",
            "queryStringParam": ".*"
        }
    ],
    "excludedQueryStringParams": [
        {
            "siteId": "",
            "queryStringParam": "gclid"
        },
        {
            "siteId": "",
            "queryStringParam": "fbclid"
        }
    ],
    "apiKey": "",
    "generatePagesWithQueryStringParams": true,
    "purgeAssetImagesWhenChanged": true,
    "refreshCacheAutomaticallyForGlobals": true,
    "refreshCacheWhenElementMovedInStructure": true,
    "refreshCacheWhenElementSavedUnchanged": false,
    "refreshCacheWhenElementSavedNotLive": false,
    "cacheNonHtmlResponses": false,
    "trackElements": true,
    "trackElementQueries": true,
    "excludedTrackedElementQueryParams": [],
    "cacheElements": true,
    "cacheElementQueries": true,
    "cacheDuration": null,
    "nonCacheableElementTypes": [],
    "sourceIdAttributes": [],
    "liveStatuses": [],
    "integrations": [
        "putyourlightson\\blitz\\drivers\\integrations\\CommerceIntegration",
        "putyourlightson\\blitz\\drivers\\integrations\\SeomaticIntegration"
    ],
    "defaultCacheControlHeader": "no-store",
    "cacheControlHeader": "public, s-maxage=31536000, max-age=0",
    "cacheControlHeaderExpired": "public, s-maxage=5, max-age=0",
    "sendPoweredByHeader": true,
    "outputComments": true,
    "refreshCacheJobPriority": 10,
    "driverJobBatchSize": 100,
    "driverJobPriority": 100,
    "queueJobTtr": 300,
    "maxRetryAttempts": 10,
    "maxUriLength": 255,
    "mutexTimeout": 1,
    "commands": [],
    "injectScriptEvent": "DOMContentLoaded",
    "injectScriptPosition": 3
}
```

#### Recommendations

- ❌ Image transforms are configured not to be generated before page load.
- ❌ One or more globals exist and refreshCacheAutomaticallyForGlobals is enabled.
- ❌ The Blitz Hints utility is reporting 61 eager-loading opportunities.
- ❌ The blitz/cache/refresh-expired console command has not been executed within the past 24 hours.
- ✅ Blitz is configured not to refresh cached pages when an element is saved but unchanged.
- ✅ Blitz is configured not to refresh cached pages when an element is saved but not live.
- ✅ The @web alias is explicitly defined.
- ✅ Queue jobs are configured not to run automatically via web requests.
- ✅ The Async Queue plugin is not installed or enabled.

#### Site Tracking [1]

- Tracked Pages: 5
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 346
	- craft\elements\Entry: 320
	- craft\elements\Asset: 25
	- verbb\formie\elements\Form: 1
- Tracked Element Queries: 15
	- craft\elements\Entry: 8
	- craft\elements\Category: 5
	- craft\elements\Asset: 1
	- verbb\formie\elements\Form: 1
- Tracked Tags: 2

#### Site Tracking [2]

- Tracked Pages: 4
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 76
	- craft\elements\Entry: 65
	- craft\elements\Asset: 11
- Tracked Element Queries: 12
	- craft\elements\Entry: 12
- Tracked Tags: 0

#### Site Tracking [3]

- Tracked Pages: 1
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 58
	- craft\elements\Entry: 48
	- craft\elements\Asset: 10
- Tracked Element Queries: 8
	- craft\elements\Category: 5
	- craft\elements\Entry: 3
- Tracked Tags: 0

#### Site Tracking [4]

- Tracked Pages: 0
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 0
- Tracked Element Queries: 0
- Tracked Tags: 0

#### Site Tracking [5]

- Tracked Pages: 0
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 0
- Tracked Element Queries: 0
- Tracked Tags: 0

#### Site Tracking [6]

- Tracked Pages: 0
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 0
- Tracked Element Queries: 0
- Tracked Tags: 0

**Comment by bencroker** [comment]
I’m unable to reproduce this, and don’t understand how this exception could happen. Can you please ensure that your vendored version of `SiteUriHelper.php` reads the same for the following lines?

https://github.com/putyourlightson/craft-blitz/blob/67bc8f6365e8519ba5e17dc122309634f2b2e541/src/helpers/SiteUriHelper.php#L461-L470

**Comment by simpleim-old** [comment]
Thanks Ben,

Can confirm that the lines are identical as above in our copy of `/vendor/putyourlightson/craft-blitz/src/helpers/SiteUriHelper.php`

**Comment by bencroker** [comment]
Is this happening in a local development environment and, if so, can you test after changing to use this code? It should dump out the value of `$siteUri` if an exception is thrown.

```php
foreach ($siteUris as $siteUri) {
    try {
        $siteUriModels[] = new SiteUriModel($siteUri);
    } catch (\Exception) {
        var_dump($siteUri);
        die();
    }
}
```

**Comment by simpleim-old** [comment]
Here's the array dumped:
```
array(3) {
  ["siteId"]=>
  int(1)
  ["uri"]=>
  string(15) "racing-calendar"
  ["cache"]=>
  NULL
}
```

That site URL and the uri is correct - matches the page that contains the template code

**Comment by bencroker** [comment]
Thanks. It’s very strange that `cache` is being set as a key when the query explicitly selects `siteId` and `uri`.

Can you please try changing line `463` to this and retest?

```php
            ->joinWith('cache', false)
```

**Comment by bencroker** [comment]
After some more testing I was able to reproduce the issue and have fixed in https://github.com/putyourlightson/craft-blitz/commit/7bc9b959eacce97c7f035bb168725047bdc9adf8 for the next release. Thanks for helping me track this down!

**Comment by bencroker** [comment]
Released in [4.23.4](https://github.com/putyourlightson/craft-blitz/releases/tag/4.23.4) and [5.9.2](https://github.com/putyourlightson/craft-blitz/releases/tag/5.9.2).
